Adding addServers method to enable pool of servers for memcache

jaskaransingh156 opened the pull request:
Adding addServers method to enable pool of servers for memcache

jaskaransingh156 commented:
It is related to the NFR mentioned in the following link:

https://github.com/phalcon/cphalcon/issues/1299

sergeyklay commented:
For indents we use tabs in zep files instead of spaces

jaskaransingh156 commented:
Thanks sergeyklay for your feedback. I have corrected the indentation and committed the changes.

sergeyklay commented on phalcon/cache/backend/memcache.zep:
```diff
@@ -117,6 +117,28 @@ class Memcache extends Backend implements BackendInterface
+         * @param  boolean persistent	 
+         * @return boolean
+	 */
+	public function addServers(var host, var port, var persistent = false)
```
I think should use such declaration:
```zephir
public function addServers(string! host, int port, boolean persistent = false) -> boolean
```

and remove `@param`s doc

See please PHP.net doc:

```php
bool Memcache::addServer ( string $host [, int $port = 11211 [, bool $persistent [, int $weight [, int $timeout [, int $retry_interval [, bool $status [, callable $failure_callback [, int $timeoutms ]]]]]]]] )
```

What with other params?

sergeyklay commented:
Update please `CHANGELOG.md`

jaskaransingh156 commented:
Thanks @sergeyklay. I have changed declaration of addServers and removed @params doc and also updated CHANGELOG.md file. Regarding other arguments of addServer method, currently only 3 options (host, port and persistent) are being accepted in the constructor of Phalcon\Cache\Backend\Memcache. This host acts as the first server which is added to the memcache pool.  So, if i provide other arguments in addServers method then constructor should be also be changed accordingly in order to maintain consistency among first and all other servers. In that case, instead of using connect method of Memcache library, we should use addServer method to add even the first server. This has been explained in the following comment also:

http://php.net/manual/en/memcache.addserver.php#Hcom104484

Please confirm if constructor of Phalcon\Cache\Backend\Memcache is to be changed so that i will make the changes accordingly.

sergeyklay commented:
Could you please rebase?

jaskaransingh156 commented:
@sergeyklay Done.

sergeyklay commented:
Thanks
